ocs_ci/utility/utils.py

```diff
@@ -1969,17 +1970,28 @@ def get_ocs_olm_operator_tags(limit=100):
     except (ValueError, TypeError):
         log.warning("Invalid ocs_version given, defaulting to ocs-registry image")
         pass
-    resp = requests.get(
-        constants.OPERATOR_CS_QUAY_API_QUERY.format(
-            tag_limit=limit,
-            image=image,
-        ),
-        headers=headers,
-    )
-    if not resp.ok:
-        raise requests.RequestException(resp.json())
-    log.debug(resp.json()["tags"])
-    return resp.json()["tags"]
+    all_tags = []
+    page = 1
+    while True:
+        log.info(f"Retrieving OCS OLM Operator tags (limit {limit}, page {page})")
+        resp = requests.get(
+            constants.OPERATOR_CS_QUAY_API_QUERY.format(
+                tag_limit=limit,
+                image=image,
+                page=page,
+            ),
+            headers=headers,
+        )
+        if not resp.ok:
+            raise requests.RequestException(resp.json())
+        tags = resp.json()["tags"]
+        if len(tags) == 0:
+            log.info("No more tags to retrieve")
+            break
+        log.debug(tags)
+        all_tags.append(tags)
+        page += 1
```

Comment:
shouldn't the while look use some kind of page <= max? how is the next loop using incremented page?

Reply:
I tested this locally and the API gracefully accepted an "empty page" and just returned an empty tag list which is what I used to determine that we reached the end.

Reply:
From my local test output:
```
Retrieving OCS OLM Operator tags (limit 100, page 8)
[]
No more tags to retrieve
```